src/commands/redis_cmd.cc

```diff
@@ -2400,19 +2409,65 @@ class CommandZAdd : public Commander {
 
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
     int ret;
+    double old_score = member_scores_[0].score;
     Redis::ZSet zset_db(svr->storage_, conn->GetNamespace());
-    rocksdb::Status s = zset_db.Add(args_[1], 0, &member_scores_, &ret);
+    rocksdb::Status s = zset_db.Add(args_[1], flags_, &member_scores_, &ret);
     if (!s.ok()) {
       return Status(Status::RedisExecErr, s.ToString());
     }
-    *output = Redis::Integer(ret);
+    bool incr = flags_.HasIncr();
+    if (incr) {
+      auto new_score = member_scores_[0].score;
+      bool nx = flags_.HasNX(), xx = flags_.HasXX(), lt = flags_.HasLT(), gt = flags_.HasGT();
+      if ((nx || xx || lt || gt) && old_score == new_score &&
```

Comment:
`incr` ,`nx`, `xx`, `lt` and `gt` are rebundant, It can be obtained directly using `Has*`.